pkg/router/template/router.go

```diff
@@ -1068,11 +1068,15 @@ func (r *templateRouter) removeRouteInternal(route *routev1.Route) {
 
 // numberOfEndpoints returns the number of endpoints
 // Must be called while holding r.lock
-func (r *templateRouter) numberOfEndpoints(id ServiceUnitKey) int32 {
+func (r *templateRouter) numberOfEndpoints(id ServiceUnitKey, port string) int32 {
 	var eps = 0
 	svc, ok := r.findMatchingServiceUnit(id)
 	if ok && len(svc.EndpointTable) > eps {
-		eps = len(svc.EndpointTable)
+		for _, ep := range svc.EndpointTable {
+			if ep.Port == port {
+				eps += 1
+			}
+		}
```

Comment:
Does this do the right thing when the route's `spec.port` is omitted (in which case the router uses all ports)?  `endpointsForAlias` checks for empty `PreferPort`, and I expect you need to do likewise here: https://github.com/openshift/router/blob/80242039851150ba43a7fdca541d3e03ddc3ba76/pkg/router/template/template_helper.go#L180-L192

Reply:
Thanks, I knew I was missing something, fixed and added more unit tests.